src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,211 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            first = last = new Node<>(null, value, null);
+            size++;
+            return true;
+        }
+        Node<T> currentNode = new Node<>(last, value, null);
+        last.next = currentNode;
+        last = currentNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            addByIndexFirstElement(value);
+            return;
+        }
+        Node<T> currentNode = iterationFromTheFirstElement(index);
+        Node<T> addNode = new Node<>(currentNode.prev, value, currentNode);
+        currentNode.prev.next = addNode;
+        currentNode.prev = addNode;
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == size - 1) {
```

Comment:
do we need this checking? let's use method  getNode(index) only

Reply:
done